src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -637,32 +641,19 @@ private void flush(boolean commitTranslog, boolean force, boolean waitIfOngoing)
             }
             try {
                 if (commitTranslog) {
-                    if (onGoingRecoveries.get() > 0) {
-                        throw new FlushNotAllowedEngineException(shardId, "Recovery is in progress, flush is not allowed");
-                    }
-
                     if (flushNeeded || force) {
                         flushNeeded = false;
+                        final long translogId;
                         try {
-                            long translogId = translogIdGenerator.incrementAndGet();
-                            translog.newTransientTranslog(translogId);
-                            indexWriter.setCommitData(Collections.singletonMap(Translog.TRANSLOG_ID_KEY, Long.toString(translogId)));
+                            translogId = translog.newTranslog();
                             logger.trace("starting commit for flush; commitTranslog=true");
```

Comment:
Can you add the `translogId` to the log message here? It makes tracking stuff down on shared filesystems much easier.